### PR TITLE
GH-40 Initial Agent View and Communication History

### PR DESCRIPTION
**Pull request by ndoschek**
#### What it does

[GH-40](https://github.com/eclipsesource/osweek-2024/issues/40) Initial Agent View and Communication History 
- Add initial `ai-history` and `ai-history-ui` extensions
- Provide CommunicationRecordingService to record agent communication
- Provide second DummyAgent for demo purposes
- Showcase recording implementation with both agents
- Add basic history UI

Co-authored-by: Olaf Lessenich <olessenich@eclipsesource.com>

Closes https://github.com/eclipsesource/osweek-2024/issues/40

#### Follow-ups

- https://github.com/eclipsesource/osweek-2024/issues/55
- https://github.com/eclipsesource/osweek-2024/issues/56
- https://github.com/eclipsesource/osweek-2024/issues/57
- https://github.com/eclipsesource/osweek-2024/issues/58
- https://github.com/eclipsesource/osweek-2024/issues/60

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-history/src/node/ai-history-backend-module.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { ContainerModule } from '@theia/core/shared/inversify';
+
+export default new ContainerModule(bind => {
+    // bind(DefaultCommunicationRecordingService).toSelf().inSingletonScope();
+    // bind(CommunicationRecordingService).toService(DefaultCommunicationRecordingService);
+});
```
Comment (sgraband) [inline packages/ai-history/src/node/ai-history-backend-module.ts:21]: Can we remove this file?

**Review by sgraband — COMMENTED** [review]
Thanks for the contribution! Works great! I just have one small comment below.

File: packages/ai-history/src/node/ai-history-backend-module.ts (hunk 1 above)
Comment (ndoschek) [reply packages/ai-history/src/node/ai-history-backend-module.ts]: Great, thank you for the review! Oh yes, that was an oversight, will push an update and rebase.

**Review by sgraband — APPROVED** [review]
Thanks! LGTM :+1:

**Pull request merged**